v1/errors/reporter.go

```diff
@@ -0,0 +1,97 @@
+// errors implements an errors/crashes reporter.
+package errors
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/getsentry/sentry-go"
+	"gopkg.in/inconshreveable/log15.v2"
+)
+
+// Reporter represents an errors reporter instance.
+type Reporter struct {
+	sentry *sentry.Client
+
+	config *Config
+}
+
+// New returns a new errors reporter instance.
+func New(config *Config) (*Reporter, error) {
+	var (
+		reporter Reporter
+		err      error
+	)
+
+	if config == nil {
+		return nil, nil
+	}
+
+	if err := config.validate(); err != nil {
+		return nil, err
+	}
+	reporter.config = config
+
+	if reporter.sentry, err = sentry.NewClient(sentry.ClientOptions{
+		Dsn:              config.DSN,
+		AttachStacktrace: true,
+	}); err != nil {
+		return nil, err
+	}
+
+	return &reporter, nil
+}
+
+// Start is a no-op operation.
+func (r *Reporter) Start(_ context.Context) error {
+	return nil
+}
+
+// Stop is a no-op operation.
+func (r *Reporter) Stop(_ context.Context) error {
+	return nil
+}
+
+// LogHandler is a log15.Handler that sends an event to Sentry if an error-level record message is logged.
+func (r *Reporter) LogHandler() log15.Handler {
+	return log15.FuncHandler(func(rec *log15.Record) error {
+		if rec.Lvl > log15.LvlError {
+			return nil
+		}
+
+		r.sentry.CaptureException(errors.New(rec.Msg), nil, sentryEventFromLogRecord(rec))
+		if r.config.Wait {
+			r.sentry.Flush(5 * time.Second)
```

Comment:
no error returned here ?

Reply:
Are you suggesting we should test that [`sentry.Flush()`](https://godoc.org/github.com/getsentry/sentry-go#Client.Flush) returns `true` and return an error otherwise?

Reply:
Sorry, I should have been clearer. I was wondering if `Flush()` returns an error, and if yes why it is not returned.

Reply:
So, what's the final suggestion then? 😏